docs/tutorial.rst

```diff
@@ -0,0 +1,1842 @@
+.. currentmodule:: reframe.core.pipeline.RegressionTest
+
+================
+ReFrame Tutorial
+================
+
+This tutorial will cover the basic concepts of ReFrame and will get you started with the framework.
+For more specific topics, you should refer to ":doc:`howto`" as well as to the ":doc:`topics`" for an in-depth understanding of some of the framework's concepts.
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 3
+
+Requirements
+============
+
+To run this tutorial you need ``docker`` for the local examples and ``docker compose`` for the examples emulating a Slurm cluster.
```

Comment:
May consider mentioning that `dockerd` actually needs to be running. This may be obvious for those that are accustomed to using Docker, but an important detail for those who are not.